### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by Robinlovelace**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,227 @@
...
+    ##     odjitter [OPTIONS] --od-csv-path <OD_CSV_PATH> --zones-path <ZONES_PATH> --output-path <OUTPUT_PATH> --max-per-od <MAX_PER_OD>
+    ## 
+    ## For more information try --help
+
+# Usage
+
+To run algorithm you need a minimum of three inputs, examples of which
+are provided in the `data/` folder of this repo:
+
+1.  A .csv file containing OD data with the first two columns containing
```
Comment (dabreegster) [inline README.md:63]: Order of columns doesn't matter. The CLI has `--origin-key` and `--destination-key` for this reason. I am philosophically weirded out by the concept of column order mattering in CSV; it's an unordered mapping from keys to values, isn't it?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,227 @@
...
+
+| geo\_code1 | geo\_code2 | all | from\_home | train | bus | car\_driver | car\_passenger | bicycle | foot | other |
+|:-----------|:-----------|----:|-----------:|------:|----:|------------:|---------------:|--------:|-----:|------:|
+| S02001616  | S02001616  |  82 |          0 |     0 |   3 |           6 |              0 |       2 |   71 |     0 |
+| S02001616  | S02001620  | 188 |          0 |     0 |  42 |          26 |              3 |      11 |  105 |     1 |
+| S02001616  | S02001621  |  99 |          0 |     0 |  13 |           7 |              3 |      15 |   61 |     0 |
+
+1.  A [.geojson
+    file](https://github.com/dabreegster/odjitter/blob/main/data/zones.geojson)
+    representing zones, with the first column matching the zone IDs in
```
Comment (dabreegster) [inline README.md:74]: "with a property (called InterZone by default) matching the zone IDs in the OD data"

GeoJSON even less so has any concept of "first column" -- it's got a JSON map of properties. There's no order

File: README.md [hunk 3]
```diff
@@ -1,3 +1,227 @@
...
+    ## "name": "road_network_min",
+    ## "crs": { "type": "name", "properties": { "name": "urn:ogc:def:crs:OGC:1.3:CRS84" } },
+    ## "features": [
+    ## { "type": "Feature", "properties": { "osm_id": "3468", "name": "Albyn Place", "highway": "tertiary", "waterway": null, "aerialway": null, "barrier": null, "man_made": null, "access": null, "bicycle": null, "service": null, "z_order": 4, "other_tags": "\"lit\"=>\"yes\",\"lanes\"=>\"3\",\"maxspeed\"=>\"20 mph\",\"sidewalk\"=>\"both\",\"lanes:forward\"=>\"2\",\"lanes:backward\"=>\"1\"" }, "geometry": { "type": "LineString", "coordinates": [ [ -3.207438, 55.9533584 ], [ -3.2065953, 55.9535098 ] ] } },
+
+The `jitter` function requires you to set the maximum number of trips
+for all trips in the jittered result. A value of 1 will create a line
+for every trip in the dataset, a value above the maximum number of trips
+in the ‘all’ column in the OD ata will result in a jittered dataset that
+has the same number of desire lines as in the input (50 in this case).
```
Comment (dabreegster) [inline README.md:108]: "Desire line" and "OD pair" are synonymous (to my knowledge). I was very confused about that early on. Could we say these two terms are equivalent somewhere further up, or maybe here?

File: README.md [hunk 4]
```diff
@@ -1,3 +1,227 @@
...
+
+With reference to the test data in this repo, you can run the `jitter`
+command line tool as follows:
+
+``` bash
+# With max-per-od of 50
```
Comment (dabreegster) [inline README.md:114]: Not sure the comment explains anything that the flag and output filename doesn't :P

**Review by dabreegster — CHANGES_REQUESTED** [review]
This is awesome! Rust crate READMEs tend to follow a certain format that might be nice to mimic later on, but this does the job of explaining the crate's purpose and how to use it. Later on, I might clarify that there's both a library that people can include it in their own code, and a CLI they can run stand-alone.

File: README.md (hunk 1 above)
Comment (Robinlovelace) [reply README.md:63]: Yes column order shouldn't matter. Happy to change that. But how should people specify which columns have the IDs in?

File: README.md (hunk 2 above)
Comment (Robinlovelace) [reply README.md:74]: :+1:

File: README.md (hunk 1 above)
Comment (dabreegster) [reply README.md:63]: By saying `--origin-key=from --destination-key=to` or whatever in the CLI. By default, the CLI defaults to geo_code1 and 2

Comment (Robinlovelace) [reply README.md:63]: Yep, that's important and worth mentioning. Will update...

File: README.md (hunk 3 above)
Comment (Robinlovelace) [reply README.md:108]: Good point. They are not quite synonymous. I plan to update this to:

```
A value of 1 will create a line for every trip in the dataset, a value above the maximum number of trips in the 'all' column in the OD ata will result in a jittered dataset that has the same number of desire lines (the geographic representation of OD pairs) as in the input (50 in this case).
```

File: README.md (hunk 4 above)
Comment (Robinlovelace) [reply README.md:114]: :+1:

**Comment by Robinlovelace** [comment]
Note: all these changes plus the source code that created some of the output in this readme are in https://github.com/dabreegster/odjitter/pull/4

**Comment by dabreegster** [comment]
Merged #4, so closing this one. Thanks!

**Pull request closed without merging**